src/qtt/measurements/post_processing/process_sawtooth_2d.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from qilib.data_set import DataSet, DataArray
+
+from qtt.measurements.post_processing import SignalProcessorInterface
+
+
+class ProcessSawtooth2D(SignalProcessorInterface):
+
+    def run_process(self, data_set: DataSet)-> DataSet:
+        """ Extracts a 2D image from a readout dot responce measured with an acquisition device.
+
+        Args:
+            data_set: The readout dot reponse data coming from the acquisition device. The data
+                      user data of the data set should contain the width and resolution settings.
+
+        Returns:
+            A data set which contains a 2D image with the charge stability diagram.
+        """
+        width_x, width_y = data_set.user_data['width']
+        resolution_x, resolution_y = data_set.user_data['resolution']
+
+        output_data_set = DataSet()
+        output_data_set.user_data = data_set.user_data
+        for _, data_array in data_set.data_arrays.items():
```

Comment:
Because you're not using the keys, then maybe it's better to iterate over the dictionary values rather than items? E.g. 
```python
for data_array in iter(data_set.data_arrays.values())
```